tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -7,6 +7,10 @@
 #include <drivers/counter.h>
 #include <ztest.h>
 #include <kernel.h>
+#ifdef CONFIG_SOC_FAMILY_NRF
```

Comment:
Why is the reason for this inclusion?
Ideally, generic samples/tests wouldn't need platform-specific inclusions, no?

Reply:
OH, I see, now, that clock_control_on() is done for nRF SoCS only.

Reply:
So, please, add some inline explanation for this, for the ease of reading ;)

Reply:
added